starter-code/src/functions-and-arrays.js

```diff
@@ -1,93 +1,459 @@
 // Iteration #1: Find the maximum
 
+function maxOfTwoNumbers(number1, number2) {
+  if (number1 >= number2) {
+    return number1;
+  } else {
+    return number2;
+  }
+}
+
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot"
+];
+
+function findLongestWord(words) {
+  var longestWord = "";
+  if (words.length === 0) {
+    return null;
+  }
+  for (var i = 0; i < words.length; i++) {
+    if (words[i].length > longestWord.length) {
+      longestWord = words[i];
+    }
+  }
+  return longestWord;
+}
 
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumArray(numbers) {
+  var summed = 0;
+  for (var i = 0; i < numbers.length; i++) {
+    summed += numbers[i];
+  }
+  return summed;
+}
 // Iteration #4: Calculate the average
+
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(numbersAvg) {
+  var summed = 0;
+  if (numbersAvg.length === 0) {
+    return null;
+  }
+  for (var i = 0; i < numbersAvg.length; i++) {
+    summed += numbersAvg[i];
+  }
+  var av = summed / numbersAvg.length;
+  return av;
+}
+
 // Level 2: Array of strings
 const wordsArr = [
-  'seat',
-  'correspond',
-  'linen',
-  'motif',
-  'hole',
-  'smell',
-  'smart',
-  'chaos',
-  'fuel',
-  'palace'
+  "seat",
+  "correspond",
+  "linen",
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace"
 ];
+function averageWordLength(wordsArr) {
+  var summ = 0;
+  if (wordsArr.length === 0) {
+    return null;
+  }
+  for (var i = 0; i < wordsArr.length; i++) {
+    summ += wordsArr[i].length;
+  }
+  var averageString = summ / wordsArr.length;
+  return averageString;
+}
```

Comment:
Also good!
If you also want to loop through the elements of an array you can also use `for.Each()` in this case for example.
In this case will make your code cleaner and it will be shorter.
As an example `worsdArr.forEach(elem=>{
summ+=elem.length
})`